python/triton/runtime/autotuner.py

```diff
@@ -184,12 +207,34 @@ def run(self, *args, **kwargs):
             if key not in self.cache:
                 # prune configs
                 used_cached_result = False
+                # Step 1: Parallel compilation
                 pruned_configs = self.prune_configs(kwargs)
+                compile_start = time.time()
+                compiled_configs = []
+                with ThreadPoolExecutor() as executor:
```

Comment:
Another thing

There is known issues with `sysconfig` library in multithreading applications: https://github.com/python/cpython/issues/92452

I've tried python 3.9 and it is affected, python 3.12 works fine, according Triton readme we target python3.8+, so I think it would be good to add a workaround:
```suggestion
                # Workaround for older python versions to safely initialize sysconfig variables
                # See https://github.com/python/cpython/issues/92452
                import sysconfig
                sysconfig.get_config_vars()
                with ThreadPoolExecutor() as executor:
```